ang/volunteer/Project.js

```diff
@@ -218,11 +224,37 @@
         valid = false;
       }
 
-      valid = (valid && $scope.validateProfileSelections());
+      valid = (valid && relationshipsValid && $scope.validateProfileSelections());
 
       return valid;
     };
 
+  /**
+   * Helper validation function.
+   *
+   * Ensures that a value is set for each required project relationship.
+   *
+   * @returns {Boolean}
+   */
+    validateRelationships = function() {
+      var isValid = true;
+
+      var requiredRelationshipTypes = ['volunteer_beneficiary', 'volunteer_manager', 'volunteer_owner'];
+
+      _.each(requiredRelationshipTypes, function(value) {
+        var thisRelType = _.find(supporting_data.values.relationship_types, function(relType) {
+          return (relType.name === value);
+        });
+
+        if (!_.isArray(relationships[thisRelType.value]) || _.isEmpty(relationships[thisRelType.value])) {
```

Comment:
Assuming that `thisRelType` will have a `value` property presents a small but acceptable risk.

The required relationship types are the machine names of the option group values installed by the extension, which are reserved. Users cannot delete them, disable them, or change the machine name without mucking around in the database. `thisRelType` should always have a `value` property in normal situations.

I'm not sure what an appropriate error message would be for the user in the case someone had mucked up the DB, so I think it's probably okay to leave this as is.